swiftide-core/src/node.rs

```diff
@@ -63,6 +63,9 @@ pub struct Node {
     /// from in bytes
     #[builder(default)]
     pub offset: usize,
+    /// Optional parent id
+    #[builder(default)]
+    pub origin_id: Option<uuid::Uuid>,
```

Comment:
Could you rename this to `parent_id`? I think that fits the comment and usecase better

Reply:
Additionaly, instead of using the builder directly, how about renaming `build_from_other` to `chunking_from`, then also setting the parent_id. Afaik it's not used anywhere else, makes it a lot more clear, and removes the need for calling the builder method.